packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -272,14 +297,25 @@ class UserSessionManager implements IUserSessionManager {
     value: Nullable<ApiObject> | Nullable<string> | undefined,
   ) {
     const entries = state.storage.entries.value;
-    const storage = entries[sessionKey]?.type as StorageType;
-    const key = entries[sessionKey]?.key as string;
-    if (isStorageTypeValidForStoringData(storage)) {
+    const storageType = entries[sessionKey]?.type as StorageType;
+    if (isStorageTypeValidForStoringData(storageType)) {
       const curStore = this.storeManager?.getStore(
-        storageClientDataStoreNameMap[storage] as string,
+        storageClientDataStoreNameMap[storageType] as string,
       );
+      const key = entries[sessionKey]?.key as string;
       if ((value && isString(value)) || isNonEmptyObject(value)) {
-        curStore?.set(key, value);
+        // if useServerSideCookie load option is set to true
+        // set the cookie from server side
+        if (state.loadOptions.value.useServerSideCookie && storageType === 'cookieStorage') {
+          const encryptedCookieValue = curStore?.encrypt(
```

Comment:
Let's move the encrypt data step to the `setServerSideCookie` as everything related to it can be together.

Reply:
Addressed